pkg/controller/linstornodeset/linstornodeset_controller.go

```diff
@@ -156,43 +158,14 @@ func (r *ReconcileLinstorNodeSet) Reconcile(request reconcile.Request) (reconcil
 		return reconcile.Result{}, err
 	}
 
-	if pns.Spec.DrbdRepoCred == "" {
-		return reconcile.Result{}, fmt.Errorf("NS Reconcile: missing required parameter drbdRepoCred: outdated schema")
-	}
-
-	if pns.Spec.SatelliteImage == "" {
-		return reconcile.Result{}, fmt.Errorf("NS Reconcile: missing required parameter satelliteImage: outdated schema")
-	}
-
-	if pns.Spec.KernelModImage == "" {
-		return reconcile.Result{}, fmt.Errorf("NS Reconcile: missing required parameter kernelModImage: outdated schema")
-	}
-
-	log := logrus.WithFields(logrus.Fields{
-		"resquestName":      request.Name,
-		"resquestNamespace": request.Namespace,
-	})
-	log.Info("NS Reconcile: reconciling LinstorNodeSet")
+	log.Debug("check if all required fields are filled")
 
-	logrus.WithFields(logrus.Fields{
-		"name":      pns.Name,
-		"namespace": pns.Namespace,
-		"spec":      fmt.Sprintf("%+v", pns.Spec),
-	}).Debug("NS Reconcile: found LinstorNodeSet")
-
-	if pns.Status.SatelliteStatuses == nil {
-		pns.Status.SatelliteStatuses = make([]*piraeusv1alpha1.SatelliteStatus, 0)
+	err = checkRequiredSpec(pns)
+	if err != nil {
+		return reconcile.Result{}, err
 	}
 
-	if pns.Spec.StoragePools == nil {
-		pns.Spec.StoragePools = &piraeusv1alpha1.StoragePools{}
-	}
-	if pns.Spec.StoragePools.LVMPools == nil {
-		pns.Spec.StoragePools.LVMPools = make([]*piraeusv1alpha1.StoragePoolLVM, 0)
-	}
-	if pns.Spec.StoragePools.LVMThinPools == nil {
-		pns.Spec.StoragePools.LVMThinPools = make([]*piraeusv1alpha1.StoragePoolLVMThin, 0)
-	}
+	log.Debug("fill in defaults in spec")
```

Comment:
This log message is kind of out of place. There are other messages when we actual perform the fill, so we can remove this.